x-pack/legacy/plugins/maps/public/layers/styles/vector/properties/dynamic_style_property.js

```diff
@@ -180,9 +185,10 @@ export class DynamicStyleProperty extends AbstractStyleProperty {
   }
 
   _pluckOrdinalStyleMetaFromFieldMetaData(fieldMetaData) {
-    const realFieldName = this._field.getESDocFieldName
-      ? this._field.getESDocFieldName()
-      : this._field.getName();
+    const realFieldName =
+      this._field instanceof ESAggMetricField
```

Comment:
Move `getRealFieldName` as a method on `Field`, and override in `ESAggMetricField` iso doing a type-check. (see above too for related comment).

Reply:
same with previous comment. I would like to work on this in a separate PR since this PR is already so large.